Composer/packages/client/src/telemetry/AppInsightsClient.ts

```diff
@@ -29,14 +29,16 @@ export default class AppInsightsClient {
     }
   }
 
-  public static trackEvent(name: string, properties: LogData) {
-    if (alwaysTrackEvents.includes(name as TelemetryEventName) || this._telemetrySettings?.allowDataCollection) {
+  public static trackEvent(name: TelemetryEventName, properties: LogData) {
+    if (alwaysTrackEvents.includes(name)) {
```

Comment:
I think this condition should be last. If telemetry is enabled, it's fine to add the event to the queue and let it propagate through the normal path.

Reply:
Good idea - will do.